src/charm.py

```diff
@@ -66,13 +98,29 @@ def _on_config_changed(self, event):
         for relation in self.mimir_worker_relations:
             for remote_unit in relation.units:
                 # todo: figure out under what circumstances this would not be routable
-                unit_ip = relation.data[remote_unit]['private-address']
+                unit_ip = relation.data[remote_unit]["private-address"]
                 hash_ring.append(unit_ip)
 
         for relation in self.mimir_worker_relations:
-            relation.data[self.app]['config'] = json.dumps(dict(self.model.config))
-            relation.data[self.app]['hash_ring'] = hash_ring
-            relation.data[self.app]['s3_storage'] = self._s3_storage
+            relation.data[self.app]["config"] = json.dumps(dict(self.model.config))
+            relation.data[self.app]["hash_ring"] = json.dumps(hash_ring)
+            relation.data[self.app]["s3_storage"] = json.dumps(self._s3_storage)
+
+    def _remote_write_endpoints_changed(self, _):
+        # TODO Update grafana-agent config file with the new external prometheus's endpoint
+        pass
+
+    def _on_ingester_changed(self, _):
```

Comment:
Nominally, the endpoint will always be proxied through the nginx here, right? So the grafana-agent remote-write target _should_ generally be the coordinator itself.

Reply:
Not sure: if we scale up/down, would we need to add/remove grafana agent targets?

Reply:
For targets, we definitely would, but that applies to.. basically every one, not just the ingester. It would basically be the same list as the hash ring memberlist (which is more or less the set of all units).

Remote write would still be the same proxied endpoint.

Reply:
Replaced with a comment in the next commit.